Agda-2.6.* dependency leads to errors

### Description

The `README.md` file states `Agda-2.6.1` as dependency but `free-compiler.cabal` defines `Agda-2.6.*`. If you have Agda-2.6.2.2 installed you will get an error trace when running `cabal new-install freec`.

### Steps to Reproduce

1. Clone the repo.
2. Install `Agda-2.6.2.2` (e.g., `cabal install Agda-2.6.2.2`)
3. Run `cabal new-install freec`.

#### Expected Behavior

Free-of-errors installation.

#### Actual Behavior

```
[ 39 of 106] Compiling FreeC.Backend.Agda.Syntax ( src/lib/FreeC/Backend/Agda/Syntax.hs, dist/build/freec-internal/FreeC/Backend/Agda/Syntax.o )

src/lib/FreeC/Backend/Agda/Syntax.hs:101:16: error:
    • Couldn't match expected type ‘ExpandedEllipsis -> t’
                  with actual type ‘LHS’
    • The first argument of ($) takes one argument,
      but its type ‘LHS’ has none
      In the expression:
        LHS argPattern [] [] $ ExpandedEllipsis NoRange 0
      In an equation for ‘lhs'’:
          lhs' = LHS argPattern [] [] $ ExpandedEllipsis NoRange 0
    • Relevant bindings include
        lhs' :: t (bound at src/lib/FreeC/Backend/Agda/Syntax.hs:101:3)
    |
101 |   lhs'       = LHS argPattern [] [] $ ExpandedEllipsis NoRange 0
    |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

src/lib/FreeC/Backend/Agda/Syntax.hs:145:14: error:
    • Couldn't match type ‘Literal -> Expr’ with ‘Expr’
      Expected type: Integer -> Expr
        Actual type: Integer -> Literal -> Expr
    • Probable cause: ‘(.)’ is applied to too few arguments
      In the expression: Lit . LitNat NoRange
      In an equation for ‘intLiteral’: intLiteral = Lit . LitNat NoRange
    |
145 | intLiteral = Lit . LitNat NoRange
    |              ^^^^^^^^^^^^^^^^^^^^

src/lib/FreeC/Backend/Agda/Syntax.hs:145:20: error:
    • Couldn't match expected type ‘Integer -> Range’
                  with actual type ‘Literal’
    • Possible cause: ‘LitNat’ is applied to too many arguments
      In the second argument of ‘(.)’, namely ‘LitNat NoRange’
      In the expression: Lit . LitNat NoRange
      In an equation for ‘intLiteral’: intLiteral = Lit . LitNat NoRange
    |
145 | intLiteral = Lit . LitNat NoRange
    |                    ^^^^^^^^^^^^^^

src/lib/FreeC/Backend/Agda/Syntax.hs:145:27: error:
    • Couldn't match expected type ‘Integer’
                  with actual type ‘Range' a1’
    • In the first argument of ‘LitNat’, namely ‘NoRange’
      In the second argument of ‘(.)’, namely ‘LitNat NoRange’
      In the expression: Lit . LitNat NoRange
    |
145 | intLiteral = Lit . LitNat NoRange
    |                           ^^^^^^^

src/lib/FreeC/Backend/Agda/Syntax.hs:149:17: error:
    • Couldn't match type ‘Literal -> Expr’ with ‘Expr’
      Expected type: String -> Expr
        Actual type: String -> Literal -> Expr
    • Probable cause: ‘(.)’ is applied to too few arguments
      In the expression: Lit . LitString NoRange
      In an equation for ‘stringLiteral’:
          stringLiteral = Lit . LitString NoRange
    |
149 | stringLiteral = Lit . LitString NoRange
    |                 ^^^^^^^^^^^^^^^^^^^^^^^

src/lib/FreeC/Backend/Agda/Syntax.hs:149:23: error:
    • Couldn't match expected type ‘String -> Range’
                  with actual type ‘Literal’
    • Possible cause: ‘LitString’ is applied to too many arguments
      In the second argument of ‘(.)’, namely ‘LitString NoRange’
      In the expression: Lit . LitString NoRange
      In an equation for ‘stringLiteral’:
          stringLiteral = Lit . LitString NoRange
    |
149 | stringLiteral = Lit . LitString NoRange
    |                       ^^^^^^^^^^^^^^^^^

src/lib/FreeC/Backend/Agda/Syntax.hs:149:33: error:
    • Couldn't match expected type ‘Data.Text.Internal.Text’
                  with actual type ‘Range' a0’
    • In the first argument of ‘LitString’, namely ‘NoRange’
      In the second argument of ‘(.)’, namely ‘LitString NoRange’
      In the expression: Lit . LitString NoRange
    |
149 | stringLiteral = Lit . LitString NoRange
    |                                 ^^^^^^^

src/lib/FreeC/Backend/Agda/Syntax.hs:156:28: error:
    • Couldn't match type ‘[]’ with ‘GHC.Base.NonEmpty’
      Expected type: Agda.Utils.List1.List1 LamBinding
        Actual type: [LamBinding' TypedBinding]
    • In the second argument of ‘Lam’, namely
        ‘(DomainFree . defaultNamedArg . mkBinder_ <$> args)’
      In the expression:
        Lam NoRange (DomainFree . defaultNamedArg . mkBinder_ <$> args)
      In an equation for ‘lambda’:
          lambda args
            = Lam NoRange (DomainFree . defaultNamedArg . mkBinder_ <$> args)
    |
156 | lambda args = Lam NoRange (DomainFree . defaultNamedArg . mkBinder_ <$> args)
    |                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

src/lib/FreeC/Backend/Agda/Syntax.hs:185:30: error:
    • Couldn't match type ‘[Expr]’ with ‘Agda.Utils.List2.List2 Expr’
      Expected type: [Expr] -> Agda.Utils.List2.List2 Expr
        Actual type: [Expr] -> [Expr]
    • In the second argument of ‘(.)’, namely
        ‘opApp' (nameNameParts $ unqualify $ op)’
      In the second argument of ‘(.)’, namely
        ‘RawApp NoRange . opApp' (nameNameParts $ unqualify $ op)’
      In the expression:
        paren . RawApp NoRange . opApp' (nameNameParts $ unqualify $ op)
    |
185 |   = paren . RawApp NoRange . opApp' (nameNameParts $ unqualify $ op)
    |                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

src/lib/FreeC/Backend/Agda/Syntax.hs:185:38: error:
    • Couldn't match type ‘GHC.Base.NonEmpty NamePart’
                     with ‘[NamePart]’
      Expected type: [NamePart]
        Actual type: NameParts
    • In the first argument of ‘opApp'’, namely
        ‘(nameNameParts $ unqualify $ op)’
      In the second argument of ‘(.)’, namely
        ‘opApp' (nameNameParts $ unqualify $ op)’
      In the second argument of ‘(.)’, namely
        ‘RawApp NoRange . opApp' (nameNameParts $ unqualify $ op)’
    |
185 |   = paren . RawApp NoRange . opApp' (nameNameParts $ unqualify $ op)
    |                                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

src/lib/FreeC/Backend/Agda/Syntax.hs:216:20: error:
    • Couldn't match expected type ‘Agda.Utils.List2.List2 Expr’
                  with actual type ‘[Expr]’
    • In the second argument of ‘RawApp’, namely
        ‘[ident "if", cond, ident "then", true, ....]’
      In the expression:
        RawApp NoRange [ident "if", cond, ident "then", true, ....]
      In an equation for ‘ifThenElse’:
          ifThenElse cond true false
            = RawApp NoRange [ident "if", cond, ident "then", ....]
    |
216 |   = RawApp NoRange [ident "if", cond, ident "then", true, ident "else", false]
    |                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

src/lib/FreeC/Backend/Agda/Syntax.hs:223:20: error:
    • Couldn't match expected type ‘Agda.Utils.List2.List2 Expr’
                  with actual type ‘[Expr]’
    • In the second argument of ‘RawApp’, namely
        ‘[ident "case", discr, ident "of", ExtendedLam NoRange alts]’
      In the expression:
        RawApp
          NoRange [ident "case", discr, ident "of", ExtendedLam NoRange alts]
      In an equation for ‘caseOf’:
          caseOf discr alts
            = RawApp NoRange [ident "case", discr, ident "of", ....]
    |
223 |   = RawApp NoRange [ident "case", discr, ident "of", ExtendedLam NoRange alts]
    |                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

src/lib/FreeC/Backend/Agda/Syntax.hs:223:54: error:
    • Couldn't match expected type ‘Expr’
                  with actual type ‘Agda.Utils.List1.List1 LamClause -> Expr’
    • Probable cause: ‘ExtendedLam’ is applied to too few arguments
      In the expression: ExtendedLam NoRange alts
      In the second argument of ‘RawApp’, namely
        ‘[ident "case", discr, ident "of", ExtendedLam NoRange alts]’
      In the expression:
        RawApp
          NoRange [ident "case", discr, ident "of", ExtendedLam NoRange alts]
    |
223 |   = RawApp NoRange [ident "case", discr, ident "of", ExtendedLam NoRange alts]
    |                                                      ^^^^^^^^^^^^^^^^^^^^^^^^

src/lib/FreeC/Backend/Agda/Syntax.hs:223:74: error:
    • Couldn't match expected type ‘Erased’
                  with actual type ‘[LamClause]’
    • In the second argument of ‘ExtendedLam’, namely ‘alts’
      In the expression: ExtendedLam NoRange alts
      In the second argument of ‘RawApp’, namely
        ‘[ident "case", discr, ident "of", ExtendedLam NoRange alts]’
    |
223 |   = RawApp NoRange [ident "case", discr, ident "of", ExtendedLam NoRange alts]
    |                                                                          ^^^^

src/lib/FreeC/Backend/Agda/Syntax.hs:231:21: error:
    • Couldn't match expected type ‘Bool -> LamClause’
                  with actual type ‘LamClause’
    • The function ‘LamClause’ is applied to four arguments,
      but its type ‘[Pattern] -> RHS -> Bool -> LamClause’ has only three
      In the expression: LamClause (lhs pat) (RHS rhs) NoWhere False
      In an equation for ‘lamClause’:
          lamClause pat rhs = LamClause (lhs pat) (RHS rhs) NoWhere False
    |
231 | lamClause pat rhs = LamClause (lhs pat) (RHS rhs) NoWhere False
    |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

src/lib/FreeC/Backend/Agda/Syntax.hs:231:32: error:
    • Couldn't match expected type ‘[Pattern]’ with actual type ‘LHS’
    • In the first argument of ‘LamClause’, namely ‘(lhs pat)’
      In the expression: LamClause (lhs pat) (RHS rhs) NoWhere False
      In an equation for ‘lamClause’:
          lamClause pat rhs = LamClause (lhs pat) (RHS rhs) NoWhere False
    |
231 | lamClause pat rhs = LamClause (lhs pat) (RHS rhs) NoWhere False
    |                                ^^^^^^^

src/lib/FreeC/Backend/Agda/Syntax.hs:231:51: error:
    • Couldn't match expected type ‘Bool’
                  with actual type ‘WhereClause' decls0’
    • In the third argument of ‘LamClause’, namely ‘NoWhere’
      In the expression: LamClause (lhs pat) (RHS rhs) NoWhere False
      In an equation for ‘lamClause’:
          lamClause pat rhs = LamClause (lhs pat) (RHS rhs) NoWhere False
    |
231 | lamClause pat rhs = LamClause (lhs pat) (RHS rhs) NoWhere False
    |                                                   ^^^^^^^

src/lib/FreeC/Backend/Agda/Syntax.hs:235:11: error:
    • Couldn't match expected type ‘ExpandedEllipsis -> LHS’
                  with actual type ‘LHS’
    • The function ‘LHS’ is applied to four arguments,
      but its type ‘Pattern -> [RewriteEqn] -> [WithExpr] -> LHS’
      has only three
      In the expression: LHS (parenIfNeeded pat) [] [] NoEllipsis
      In an equation for ‘lhs’:
          lhs pat = LHS (parenIfNeeded pat) [] [] NoEllipsis
    |
235 | lhs pat = LHS (parenIfNeeded pat) [] [] NoEllipsis
    |           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

src/lib/FreeC/Backend/Agda/Syntax.hs:260:50: error:
    • Couldn't match type ‘[NamedArg Binder]’
                     with ‘GHC.Base.NonEmpty (NamedArg Binder)’
      Expected type: Agda.Utils.List1.List1 (NamedArg Binder)
        Actual type: [NamedArg Binder]
    • In the second argument of ‘TBind’, namely
        ‘(map visibleArg types)’
      In the second argument of ‘($)’, namely
        ‘TBind NoRange (map visibleArg types) expr’
      In the expression:
        DomainFull $ TBind NoRange (map visibleArg types) expr
    |
260 | binding types expr = DomainFull $ TBind NoRange (map visibleArg types) expr
    |                                                  ^^^^^^^^^^^^^^^^^^^^

src/lib/FreeC/Backend/Agda/Syntax.hs:275:50: error:
    • Couldn't match type ‘[TypedBinding' Expr]’
                     with ‘GHC.Base.NonEmpty TypedBinding’
      Expected type: Telescope1
        Actual type: [TypedBinding' Expr]
    • In the first argument of ‘Pi’, namely ‘(binder : binders)’
      In the expression: Pi (binder : binders) expr'
      In an equation for ‘pi’:
          pi decls expr
            | (Pi binders expr') <- expr = Pi (binder : binders) expr'
            | otherwise = Pi [binder] expr
            where
                binder
                  = TBind NoRange (map hiddenArg decls) $ Underscore NoRange Nothing
    |
275 | pi decls expr | (Pi binders expr') <- expr = Pi (binder : binders) expr'
    |                                                  ^^^^^^^^^^^^^^^^

src/lib/FreeC/Backend/Agda/Syntax.hs:275:59: error:
    • Couldn't match type ‘GHC.Base.NonEmpty TypedBinding’
                     with ‘[TypedBinding' Expr]’
      Expected type: [TypedBinding' Expr]
        Actual type: Telescope1
    • In the second argument of ‘(:)’, namely ‘binders’
      In the first argument of ‘Pi’, namely ‘(binder : binders)’
      In the expression: Pi (binder : binders) expr'
    |
275 | pi decls expr | (Pi binders expr') <- expr = Pi (binder : binders) expr'
    |                                                           ^^^^^^^

src/lib/FreeC/Backend/Agda/Syntax.hs:276:32: error:
    • Couldn't match type ‘[TypedBinding' Expr]’
                     with ‘GHC.Base.NonEmpty TypedBinding’
      Expected type: Telescope1
        Actual type: [TypedBinding' Expr]
    • In the first argument of ‘Pi’, namely ‘[binder]’
      In the expression: Pi [binder] expr
      In an equation for ‘pi’:
          pi decls expr
            | (Pi binders expr') <- expr = Pi (binder : binders) expr'
            | otherwise = Pi [binder] expr
            where
                binder
                  = TBind NoRange (map hiddenArg decls) $ Underscore NoRange Nothing
    |
276 |               | otherwise = Pi [binder] expr
    |                                ^^^^^^^^

src/lib/FreeC/Backend/Agda/Syntax.hs:278:27: error:
    • Couldn't match type ‘[NamedArg Binder]’
                     with ‘GHC.Base.NonEmpty (NamedArg Binder)’
      Expected type: Agda.Utils.List1.List1 (NamedArg Binder)
        Actual type: [NamedArg Binder]
    • In the second argument of ‘TBind’, namely ‘(map hiddenArg decls)’
      In the expression: TBind NoRange (map hiddenArg decls)
      In the expression:
        TBind NoRange (map hiddenArg decls) $ Underscore NoRange Nothing
    |
278 |   binder = TBind NoRange (map hiddenArg decls) $ Underscore NoRange Nothing
    |                           ^^^^^^^^^^^^^^^^^^^
cabal: Failed to build lib:freec-internal from free-compiler-0.3.0.0 (which is
required by exe:freec from free-compiler-0.3.0.0). See the build log above for
details.
```

### Versions

 - Git commit: 6931b9ca652a185a92dd824373f092823aea4ea9
 - `ghc --version`: 8.6.5
 - `cabal --version`: 3.4.0.0
 - `coqc --version`: 8.12.2
 - OS: macOS Monterey 12.3.1

### Known Workarounds

```diff
diff --git a/free-compiler.cabal b/free-compiler.cabal
index 74adedd6..d3c0d9f0 100644
--- a/free-compiler.cabal
+++ b/free-compiler.cabal
@@ -56,7 +56,7 @@ flag PolysemyPluginEnabled
 common deps
   build-depends:     base                        ==4.12.*
                      -- ASTs:
-                   , Agda                        ==2.6.*
+                   , Agda                        ==2.6.1
```

--- a/free-compiler.cabal
+++ b/free-compiler.cabal
@@ -56,7 +56,7 @@ flag PolysemyPluginEnabled
 common deps
   build-depends:     base                        ==4.12.*
                      -- ASTs:
-                   , Agda                        ==2.6.*
+                   , Agda                        ==2.6.1
                    , haskell-src-exts            ==1.23.*
                    , language-coq                ==0.4.*
                      -- Configuration:

